### PR TITLE
Fix warning

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

fix https://github.com/istio/istio/issues/50805

### CODE REVIEW

File: pkg/config/analysis/msg/messages.yaml [hunk 1]
```diff
@@ -298,7 +298,7 @@ messages:
   - name: "ServiceEntryAddressesRequired"
     code: IST0134
     level: Warning
-    description: "Virtual IP addresses are required for ports serving TCP (or unset) protocol"
+    description: "Virtual IP addresses are required for ports serving TCP (or unset) protocol when ISTIO_META_DNS_AUTO_ALLOCATE is not set on a proxy"
```
Comment (hanxiaop) [inline pkg/config/analysis/msg/messages.yaml:301]: I think this doesn't fix the issue as users will still get warnings when the env variable is enabled.

https://github.com/istio/istio/blob/master/pkg/config/analysis/analyzers/serviceentry/protocoladdresses.go#L66 the analyzer already has a specific logic to it, but unfortunately the schema validation throws the warning.

**Review by hanxiaop — COMMENTED** [review]
The description link seems to be wrong, I assume it's for https://github.com/istio/istio/issues/50805

File: pkg/config/analysis/msg/messages.yaml (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/config/analysis/msg/messages.yaml:301]: It doesnot matter, we cannot know whether proxy has that set. It is just a hint

Comment (hzxuzhonghu) [reply pkg/config/analysis/msg/messages.yaml:301]: https://github.com/istio/istio/blob/master/pkg/config/analysis/analyzers/serviceentry/protocoladdresses.go#L66 only check meshconfig setting,  it is still not enough

Comment (hanxiaop) [reply pkg/config/analysis/msg/messages.yaml:301]: Makes sense, this is better than the message before

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**